crates/bevy_render/src/view/visibility/mod.rs

```diff
@@ -68,9 +76,60 @@ impl VisibleEntities {
     }
 }
 
+/// Tracks which [`Entities`](Entity) have which meshes for entities whose [`Aabb`]s are managed by
+/// the [`calculate_bounds`] and [`update_bounds`] systems. This is needed because `update_bounds`
+/// recomputes `Aabb`s for entities whose mesh has been mutated. These mutations are visible via
+/// [`AssetEvent<Mesh>`](AssetEvent) which tells us which mesh was changed but not which entities
+/// have that mesh.
+#[derive(Debug, Default, Clone)]
+pub struct EntityMeshMap {
+    entities_with_mesh: HashMap<Handle<Mesh>, SmallVec<[Entity; 1]>>,
+    mesh_for_entity: HashMap<Entity, Handle<Mesh>>,
+}
+
+impl EntityMeshMap {
+    /// Register the passed `entity` as having the passed `mesh_handle`.
+    fn register(&mut self, entity: Entity, mesh_handle: &Handle<Mesh>) {
+        // Note that this list can have duplicates if an entity is registered for a mesh multiple
+        // times. This should be rare and only cause an additional `Aabb.clone()` in
+        // `update_bounds` so it is preferable to a `HashSet` for now.
+        self.entities_with_mesh
+            .entry(mesh_handle.clone_weak())
+            .or_default()
+            .push(entity);
+        self.mesh_for_entity
+            .insert(entity, mesh_handle.clone_weak());
+    }
+
+    /// Deregisters the mapping between an `Entity` and `Mesh`. Used so [`update_bounds`] can
+    /// track which mappings are still active so `Aabb`s are updated correctly.
+    fn deregister(&mut self, entity: Entity) {
+        let mut inner = || {
+            let mesh = self.mesh_for_entity.remove(&entity)?;
+            let entities = self.entities_with_mesh.get_mut(&mesh)?;
+
+            // There could be duplicate entries in here if an entity was registered with a mesh
+            // twice. It's important to remove all references so that if an entity gets a new mesh
```

Comment:
I wanted this change to be consistent with the comment in `register` but I didn't save the file. I'll commit this locally because this causes a reflow of the comment but I'll wait until things are approved before worrying to save on CI cycles and notifications
```suggestion
            // multiple times. It's important to remove all references so that if an entity gets a new mesh
```